controllers/util/events.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The RamenDR authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
```

Comment:
(note) We may want to add a line that we are sourcing this from odf/ocs/rook as the case may be for future reference and reconciling the code from those repositories when changes occur in them.

Reply:
While the changes are mainly based out of ocs, some changes were made for ramen. For instance, ocs has the function ReportIfNotPresent as a method that belongs to the struct (or class in a more generic term) EventReporter. But in this case, its an independent function that can be consumed by anyone.

Also, in future when we have to manage the events for reconciler actions of other controllers (such as DRPC, DRPolicy etc), we may have to make further modifications.

I can still go ahead and add a comment saying the implementation is mainly based out of how ocs operator handles the events.

Reply:
Let's add the note anyway, helps future implementers.

Reply:
ACK